zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -81,6 +88,62 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Describes the key inputs and UFVK for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HdSeedAccount(
+    pub HdSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
```

Comment:
If you insist on this having public fields, I will insist on this being a struct with named fields, not a tuple.

Reply:
It's not that I insist. When I said "if you insist, I'll fold" I meant I'd give in... not that I'd quit. Just to be clear. :)

Reply:
In general we prefer structs with named private fields, a constructor, and accessor methods. In this case in particular it doesn't really make sense to have the fields be public (which also means that they're mutable relative to one another, when you don't really want mutation at all.)

Reply:
In making the requested change, I lost the ability to do pattern matching. This no longer works:
```rs
if let Account::Zip32(HdSeedAccount(_, account_index, ufvk)) = account {
```

Reply:
You can do pattern matching to destructure structs:
```rust
if let Account::Zip32(HdSeedAccount { account_index, ufvk, .. }) = account {
```

Reply:
That doesn't appear to work. And I'm not sure how it would when fields are private. But it's not blocking, so sometime maybe we can review this.